src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,293 @@
+            if (this.iframe && url) {
+                this.prepIframeForDisplay(url)
+                this.setTimerForPopup(target)
+            }
+        }
+    }
+
+    private mouseOutListener = (e: MouseEvent) => {
+        const nextTarget = e.relatedTarget as HTMLElement
+        if (this.shouldRemoveOnMouseOut(nextTarget)) {
+            this.hidePreview()
+        }
+    }
+
+    private hidePreview() {
+        this.hoveredElement = null
+        this.clearPopupTimeout()
+        this.resetIframeForNextHover()
+        this.destroyPopper()
+    }
+
+    private shouldRemoveOnMouseOut(nextTarget: HTMLElement) {
+        const isNotHoveringOverTarget = nextTarget !== this.hoveredElement
+        const isNotHoveringOverIframe = nextTarget !== this.iframe
+        return isNotHoveringOverIframe && isNotHoveringOverTarget
+    }
+
+    private setTimerForPopup(target: HTMLElement) {
+        if (!this.popupTimeout) {
+            this.popupTimeout = window.setTimeout(() => {
+                if (this.iframe) {
+                    this.showPreview()
+                    this.makePopper(target)
+                }
+            }, this.popupTimeoutDuration)
+        }
+    }
+
+    private getTargetInnerText(target: HTMLElement) {
+        // remove '#' for page tags
+        const isPageRefTag = this.isTargetPageRefTag(target)
+        return isPageRefTag ? target.innerText.slice(1) : target.innerText
+    }
+
+    private isTargetPageRefTag(target: HTMLElement) {
+        return target.classList.contains('rm-page-ref-tag')
+    }
+
+    private isTargetPageRef(target: HTMLElement) {
+        return target.classList.contains('rm-page-ref')
+    }
+
+    private destroyPopper() {
+        if (this.popper) {
+            this.popper.destroy()
+            this.popper = null
+        }
+    }
+
+    private resetIframeForNextHover() {
+        if (this.iframe) {
+            this.scrollToTopOnMouseOut()
+            this.iframe.style.pointerEvents = 'none'
+            this.iframe.style.opacity = '0'
+            this.iframe.style.height = '0'
+            this.iframe.style.width = '0'
+        }
+    }
+
+    private scrollToTopOnMouseOut() {
+        if (this.iframe?.contentDocument) {
+            // scroll to top when removed, so the next popup is not scrolled
+            const scrollContainer = this.iframe.contentDocument.querySelector('.roam-center > div')
+            if (scrollContainer) {
+                scrollContainer.scrollTop = 0
+            }
+        }
+    }
+
+    private showPreview() {
+        if (this.iframe) {
+            this.iframe.style.opacity = '1'
+            this.iframe.style.pointerEvents = 'all'
+        }
+    }
+    private prepIframeForDisplay(url: string) {
+        if (!this.iframe) {
+            return
+        }
+        const visibleIframe = this.getVisibleIframeByUrl(url)
+        if (visibleIframe) {
+            // if visible, just return the iframe
+            return
+        }
```

Comment:
should use class iframe instance

Reply:
@Stvad will not work -- mouse events get called 3 times